frontend/src/components/pages/search/components/FilterBar/ShowFilters.tsx

```diff
@@ -1,17 +1,56 @@
 import { SelectableDropdown } from 'components/pages/search/components/FilterBar/SelectableDropdown';
 import Field from 'components/pages/search/components/FilterBar/Field';
 import React from 'react';
-import { FilterState, Option } from '../../../../../modules/filters/interface';
+import { useIntl } from 'react-intl';
+import { DateFilter, FilterState, Option } from '../../../../../modules/filters/interface';
+import InputDateWithMagnifier from '../InputDateWithMagnifier';
 
 interface Props {
   item: FilterState;
   setFilterSelectedOptions: (filterId: string, options: Option[]) => void;
   hideLabel?: boolean;
+  dateFilter: DateFilter;
+  setDateFilter: any;
 }
 
-const ShowFilters: React.FC<Props> = ({ item, setFilterSelectedOptions, hideLabel = false }) => {
+const ShowFilters: React.FC<Props> = ({
+  item,
+  setFilterSelectedOptions,
+  hideLabel = false,
+  dateFilter,
+  setDateFilter,
+}) => {
+  const intl = useIntl();
+
+  if (item !== undefined && item.id === 'date-filter') {
+    return (
+      <div className="flex flex-col mt-4 desktop:mt-0 desktop:ml-5">
+        <div className="font-bold mb-2 text-lg">{intl.formatMessage({ id: 'search.diary' })}</div>
+        <InputDateWithMagnifier
+          value={dateFilter.beginDate}
+          onChange={event => {
+            setDateFilter({
+              beginDate: event.target.value,
+              endDate: dateFilter.endDate,
+            });
+          }}
+          placeholder={intl.formatMessage({ id: 'search.beginDateFilter' })}
+        />
+        <InputDateWithMagnifier
+          value={dateFilter.endDate}
+          onChange={event => {
+            setDateFilter({
+              beginDate: dateFilter.beginDate,
+              endDate: event.target.value,
+            });
+          }}
+          placeholder={intl.formatMessage({ id: 'search.endDateFilter' })}
+        />
+      </div>
+    );
+  }
   // The API can send empty item
-  if (item.label === '' || item.options.length === 0) {
+  if (item === undefined || item.label === '' || item.options.length === 0) {
     return null;
   }
```

Comment:
Maybe, you can keep this early return at the top of the function and thus remove the test `item !== undefined` of your condition?